pkg/proofs/prefixproof.go

```diff
@@ -58,7 +61,9 @@ type PrefixTree struct {
 // ascending by steps.Step.Vrf_output and that coPathNodes is sorted ascending
 // too. prefix will be initially empty and reflects the current position in the
 // prefix tree.
-//@ ensures err != nil ==> tree != nil && tree.Inv()
+// @ requires forall i int :: { steps[i] } 0 <= i && i < len(steps) ==> acc(&steps[i], _) && acc(steps[i].Inv(), _)
+// @ requires forall i int :: { coPathNodes[i] } 0 <= i && i < len(coPathNodes) ==> acc(&coPathNodes[i], _)
+// @ ensures err == nil ==> tree != nil && tree.Inv()
 func ToTreeRecursive(prefix []bool, steps []CompleteBinaryLadderStep, coPathNodes []NodeValue) (tree *PrefixTree, nextSteps []CompleteBinaryLadderStep, nextNodes []NodeValue, err error) {
```

Comment:
```
Precondition of call ToTreeRecursive([]bool{}, steps, prf.Elements) might not hold. 
Permission to acc(steps[i].Inv(), _) might not suffice.
```

Why do these permissions not carry over from the postcondition of `CombineResults`?

https://github.com/felixlinker/keytrans-verification/blob/d9623253007f10b6e3e676715edd5e51489ef59c/pkg/proofs/prefixproof.go#L211-L218
https://github.com/felixlinker/keytrans-verification/blob/d9623253007f10b6e3e676715edd5e51489ef59c/pkg/proofs/proofs.go#L127-L128

Reply:
Fixed in 4c76db4ddc85259f121659f70408e9ef82630a6c.